src/redisearch_rs/inverted_index/src/tests.rs

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+use std::mem::ManuallyDrop;
+
+use ffi::t_docId;
+
+use crate::{RSAggregateResult, RSIndexResult, RSIndexResultData, RSResultType, RSResultTypeMask};
+
+#[unsafe(no_mangle)]
+pub extern "C" fn IndexResult_ConcatMetrics(
+    _parent: *mut RSIndexResult,
+    _child: *mut RSIndexResult,
+) {
+    // Do nothing since the code will call this
+}
+
+impl RSAggregateResult {
+    fn get(&self, index: usize) -> Option<&RSIndexResult> {
+        if index > self.num_children as _ {
+            return None;
+        }
+
+        let child_ptr = unsafe { *self.children.add(index) };
+
+        if child_ptr.is_null() {
+            None
+        } else {
+            let child = unsafe { &*child_ptr };
+            Some(child)
+        }
+    }
+}
+
+impl RSIndexResult {
+    fn union(doc_id: t_docId, cap: usize, weight: f64) -> Self {
```

Comment:
Why have this in the tests? We already have such similar constructors.

Reply:
I was going to move it in one of the later PRs to keep the public interface as small as possible

Reply:
It got moved in https://github.com/RediSearch/RediSearch/pull/6490/commits/e9640e7f0493fe063f61d386859f249d7d177902